internal/ethapi/transaction_args.go

```diff
@@ -111,7 +111,7 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 			AccessList:           args.AccessList,
 		}
 		pendingBlockNr := rpc.BlockNumberOrHashWithNumber(rpc.PendingBlockNumber)
-		estimated, err := DoEstimateGas(ctx, b, callArgs, pendingBlockNr, b.RPCGasCap())
+		estimated, err := DoEstimateGas(ctx, b, callArgs, pendingBlockNr, b.RPCGasCap(), false)
```

Comment:
Hi @holiman It's here, in the eth_sendTrasaction RPC, if the given tx doesn't set `gasLimit`, it will be set by `DoEstimateGas`, and in that case, I think we need to calculate the gas also by baseFee